cirq-core/cirq/work/observable_measurement.py

```diff
@@ -531,7 +531,7 @@ def measure_grouped_settings(
     for max_setting, param_resolver in itertools.product(
         grouped_settings.keys(), study.to_resolvers(circuit_sweep)
     ):
-        circuit_params = param_resolver.param_dict
+        circuit_params = dict(param_resolver.param_dict)
```

Comment:
Let's file an issue about fixing the types so we can avoid the copies here and in `sample_expectation_values` below. I think these should be fixable without worrying too much about compatibility, since AFAICT the dicts are not exposed anywhere in a way that is intended to be mutable, but we can deal with that in a future PR.

Reply:
Opened #5554.